stdpopsim/models.py

```diff
@@ -242,40 +204,53 @@ def debug(self, out_file=sys.stdout):
             demographic_events=self.demographic_events)
         dd.print_history(out_file)
 
-    # TODO deprecate/remove this, as the recommended interface is now 'run'.
-    def asdict(self):
-        return {
-            "population_configurations": self.population_configurations,
-            "migration_matrix": self.migration_matrix,
-            "demographic_events": self.demographic_events}
+    def get_samples(self, *args):
+        """
+        Returns a list of msprime.Sample objects as described by the args and
+        keyword args. Positional arguments are interpreted as the number of
+        samples to take from the given population.
+        """
+        samples = []
+        for pop_index, n in enumerate(args):
+            samples.extend([msprime.Sample(pop_index, time=0)] * n)
+        return samples
 
-    def run(self, chromosome, samples):
+    def run(self, contig, samples):
         """
-        Runs this model for the specified chromosome (defining the recombination
+        Runs this model for the specified contig (defining the recombination
         map and mutation rate) and samples.
         """
         ts = msprime.simulate(
             samples=samples,
-            recombination_map=chromosome.recombination_map,
-            mutation_rate=chromosome.mutation_rate,
+            recombination_map=contig.recombination_map,
+            mutation_rate=contig.mutation_rate,
             population_configurations=self.population_configurations,
             migration_matrix=self.migration_matrix,
             demographic_events=self.demographic_events)
         return ts
 
 
-def all_models():
-    """
-    Returns the list of all Model classes that are defined within the stdpopsim
-    module.
-    """
-    # TODO remove this function and change to model of "registering" models
-    # like we do for other objects.
-    ret = []
-
-    for scls in Model.__subclasses__():
-        for cls in scls.__subclasses__():
-            mod = inspect.getmodule(cls).__name__
-            if mod.startswith("stdpopsim"):
-                ret.append(cls())
-    return ret
+# Resuable generic population
+_pop0 = Population(name="pop0", description="Generic population")
```

Comment:
so this is a bit strange. will we have a `_pop1` when we add a generic IM model?

Reply:
Yeah, I guess so. This would have a different description to make it clear which population was simulated (this class really just exists to create metadata for the downstream TreeSequence.populations table).